appbars/src/main/java/com/decathlon/vitamin/appbars/PrimaryActionModeCallback.kt

```diff
@@ -0,0 +1,46 @@
+package com.decathlon.vitamin.appbars
+
+import android.view.Menu
+import android.view.MenuItem
+import androidx.annotation.MenuRes
+import androidx.appcompat.view.ActionMode
+
+class PrimaryActionModeCallback(
+    @MenuRes val menuResId: Int,
+    val onActionItemClickListener: ((item: MenuItem) -> Unit)? = null,
+    val onActionModeFinishedListener: (() -> Unit)? = null
+) : ActionMode.Callback {
+    private var actionMode: ActionMode? = null
+
+    override fun onCreateActionMode(mode: ActionMode, menu: Menu): Boolean {
+        this.actionMode = mode
+        mode.menuInflater.inflate(menuResId, menu)
+        return true
+    }
+
+    override fun onPrepareActionMode(mode: ActionMode, menu: Menu): Boolean {
+        return false
+    }
+
+    override fun onDestroyActionMode(mode: ActionMode) {
+        onActionModeFinishedListener?.invoke()
+        this.actionMode = null
+    }
+
+    override fun onActionItemClicked(mode: ActionMode, item: MenuItem): Boolean {
+        onActionItemClickListener?.invoke(item)
+        mode.finish()
```

Comment:
Is there a specific reason to finish the `actionMode` on any item click?
I think it can be confusing for the library user if he doesn't want to do that, it could be hard to find that responsible line. If he want to do that, it could still call the `finishActionMode` when the `onActionItemClickListener` is called.

Reply:
It is the default behavior of a contextual toolbar. You select multiple items in a list, you have contextualized actions in the toolbar and when you click on one action, it disappears and the action is applied on the list. I think that should the default behavior but I can add a boolean in the constructor of the class to apply this behavior or not maybe?